nifi-external/nifi-kafka-connect/nifi-kafka-connector/src/main/java/org/apache/nifi/kafka/connect/StatelessNiFiCommonConfig.java

```diff
@@ -0,0 +1,267 @@
+        return getString(EXTENSIONS_DIRECTORY);
+    }
+
+    public String getWorkingDirectory() {
+        return getString(WORKING_DIRECTORY);
+    }
+
+    public String getDataflowName() {
+        return getString(DATAFLOW_NAME);
+    }
+
+    public String getKrb5File() {
+        return getString(KRB5_FILE);
+    }
+
+    public String getNexusBaseUrl() {
+        return getString(NEXUS_BASE_URL);
+    }
+
+    public String getDataflowTimeout() {
+        return getString(DATAFLOW_TIMEOUT);
+    }
+
+    public String getKeystoreFile() {
+        return getString(KEYSTORE_FILE);
+    }
+
+    public String getKeystoreType() {
+        return getString(KEYSTORE_TYPE);
+    }
+
+    public String getKeystorePassword() {
+        return getOptionalPassword(KEYSTORE_PASSWORD);
+    }
+
+    public String getKeystoreKeyPassword() {
+        return getOptionalPassword(KEY_PASSWORD);
+    }
+
+    public String getTruststoreFile() {
+        return getString(TRUSTSTORE_FILE);
+    }
+
+    public String getTruststoreType() {
+        return getString(TRUSTSTORE_TYPE);
+    }
+
+    public String getTruststorePassword() {
+        return getOptionalPassword(TRUSTSTORE_PASSWORD);
+    }
+
+    public String getSensitivePropsKey() {
+        return getOptionalPassword(SENSITIVE_PROPS_KEY);
+    }
+
+    /**
+     * Populates the properties with the data flow definition params.
+     *
+     * @param dataflowDefinitionProperties The properties to populate.
+     */
+    public void provideFlowDefinition(Map<String, String> dataflowDefinitionProperties) {
+        String configuredFlowSnapshot = getString(FLOW_SNAPSHOT);
+        if (configuredFlowSnapshot.startsWith("http://") || configuredFlowSnapshot.startsWith("https://")) {
+            logger.debug("Configured Flow Snapshot appears to be a URL. Will use {} property to configured Stateless NiFi", StatelessNiFiCommonConfig.BOOTSTRAP_SNAPSHOT_URL);
+            dataflowDefinitionProperties.put(StatelessNiFiCommonConfig.BOOTSTRAP_SNAPSHOT_URL, configuredFlowSnapshot);
+        } else if (configuredFlowSnapshot.trim().startsWith("{")) {
+            logger.debug("Configured Flow Snapshot appears to be JSON. Will use {} property to configured Stateless NiFi", StatelessNiFiCommonConfig.BOOTSTRAP_SNAPSHOT_CONTENTS);
+            dataflowDefinitionProperties.put(StatelessNiFiCommonConfig.BOOTSTRAP_SNAPSHOT_CONTENTS, configuredFlowSnapshot);
+        } else {
+            logger.debug("Configured Flow Snapshot appears to be a File. Will use {} property to configured Stateless NiFi", StatelessNiFiCommonConfig.BOOTSTRAP_SNAPSHOT_FILE);
+            final File flowSnapshotFile = new File(configuredFlowSnapshot);
+            dataflowDefinitionProperties.put(StatelessNiFiCommonConfig.BOOTSTRAP_SNAPSHOT_FILE, flowSnapshotFile.getAbsolutePath());
+        }
+    }
+
+    /**
+     * @return The parameter overrides of the flow.
+     */
+    public List<ParameterOverride> provideParameterOverrides() {
```

Comment:
Recommend adjusting the method name since `provide` is not commonly used.
```suggestion
    public List<ParameterOverride> getParameterOverrides() {
```